### DISCUSSION TITLE
Enabling (open AP mode) CYW43

### DISCUSSION THREAD

**Discussion by BMOC1** [micropython/micropython#11477] [Using MicroPython] [answered]
I am running MicroPython version : 3.4.0: MicroPython v1.20.0 on 2023-04-26

I stumbled across the following URL which led me to believe there was a bug not allowing open AP mode : 
https://github.com/georgerobotics/cyw43-driver/issues/27

I don't know if the MicroPython version i am running still has this bug, or if my config field is incorrect. (ie error in my coding)

I was attempting to enable AP mode with no security.  Open Mode.

```
wlan = network.WLAN(network.AP_IF)
wlan.disconnect()
wlan.config(eesid=dev_short, authmode=0)
```
Which then yields the following error : ValueError: unknown config param
I have also tried auth_type 

Thank you ahead of time.  At this point, i am at a dead stop.

**Comment by jimmo** [comment]
> wlan.config(eesid=dev_short, authmode=0)
 
> I have also tried auth_type

As per the documentation (see https://docs.micropython.org/en/latest/library/network.WLAN.html#network.WLAN.config) the argument is `security`. (Set it to zero for open).

**Reply by BMOC1** [reply]
> > wlan.config(eesid=dev_short, authmode=0)
> 
> > I have also tried auth_type
> 
> As per the documentation (see https://docs.micropython.org/en/latest/library/network.WLAN.html#network.WLAN.config) the argument is `security`. (Set it to zero for open).

I had tried that... Here is what happens when i issue that config :

```
wlan = network.WLAN(network.AP_IF)
wlan.config(security=0)

Traceback (most recent call last):
  File "<stdin>", line 261, in <module>
ValueError: unknown config param
```

**Reply by robert-hh** [reply]
```
wlan = network.WLAN(network.AP_IF)
wlan.config(security=0)
```
works here on a PYBD-SF6. Which board do you use?

**Reply by BMOC1** [reply]
MicroPython v1.20.0 on 2023-04-26; Raspberry Pi Pico W with RP2040

**Reply by robert-hh** [reply, marked as answer]
The sequence below works here as well with a Pico W with the same firmware version `MicroPython v1.20.0 on 2023-04-26; Raspberry Pi Pico W with RP2040`
```
import network
wlan = network.WLAN(network.AP_IF)
wlan.config(security=0)
```
If you just run these three lines, does it fail as well? What else is in your scripts?

**Reply by BMOC1** [reply]
I feel like such a dork.  I spent so much time on this one little issue, and tried so many different code scenarios, i should have just gone back to the basics.  I've been programming for 30+ years, but i am new to python.   Thank you **robert-hh**  -  I am still having the issue on my code, ill have to debug it and try and figure out what the trigger is.  The program starts out in station mode, tries configured ssid, and if it fails after so many attempts, searches for an open network to connect to, and if that fails, switches to AP mode, allowing a user to get back into the device if all else fails.  This is all done asynchronously so the program can attend to other tasks.

I've written over 500 lines of code for this program so far, and this is the first situation where i reached out.. I think when i found the AP bug and after so many failed attempts - also adding to the confusion, i figured out some times (after stop/starting code over and over) the pico can get into a weird **network** state and required a power cycle.  All those things coupled together - made my head garbled!   Thanks again!
